tests/test_app/components.py

```diff
@@ -223,3 +235,126 @@ def _render_items(items, toggle_item):
             for item in items
         ]
     )
+
+
+@component
+def view_to_component_sync_func():
+    return view_to_component(views.view_to_component_sync_func)
+
+
+@component
+def view_to_component_async_func():
+    return view_to_component(views.view_to_component_async_func)
+
+
+@component
+def view_to_component_sync_class():
+    return view_to_component(views.ViewToComponentSyncClass)
+
+
+@component
+def view_to_component_async_class():
+    return view_to_component(views.ViewToComponentAsyncClass)
+
+
+@component
+def view_to_component_template_view_class():
+    return view_to_component(views.ViewToComponentTemplateViewClass)
+
+
+@component
+def view_to_component_sync_func_compatibility():
+    return html.div(
+        {"id": inspect.currentframe().f_code.co_name},
+        view_to_component(
+            views.view_to_component_sync_func_compatibility, compatibility=True
+        ),
+        html.hr(),
+    )
+
+
+@component
+def view_to_component_async_func_compatibility():
+    return html.div(
+        {"id": inspect.currentframe().f_code.co_name},
+        view_to_component(
+            views.view_to_component_async_func_compatibility, compatibility=True
+        ),
+        html.hr(),
+    )
+
+
+@component
+def view_to_component_sync_class_compatibility():
+    return html.div(
+        {"id": inspect.currentframe().f_code.co_name},
+        view_to_component(
+            views.ViewToComponentSyncClassCompatibility, compatibility=True
+        ),
+        html.hr(),
+    )
+
+
+@component
+def view_to_component_async_class_compatibility():
+    return html.div(
+        {"id": inspect.currentframe().f_code.co_name},
+        view_to_component(
+            views.ViewToComponentAsyncClassCompatibility, compatibility=True
+        ),
+        html.hr(),
+    )
+
+
+@component
+def view_to_component_template_view_class_compatibility():
+    return html.div(
+        {"id": inspect.currentframe().f_code.co_name},
+        view_to_component(
+            views.ViewToComponentTemplateViewClassCompatibility, compatibility=True
+        ),
+        html.hr(),
+    )
+
+
+@component
+def view_to_component_script():
+    return view_to_component(views.view_to_component_script)
+
+
+@component
+def view_to_component_request():
+    request, set_request = hooks.use_state(None)
+
+    @hooks.use_effect
+    async def change_request():
+        await asyncio.sleep(3)
```

Comment:
Instead of depending on these 3 second delays, can we do something a little more deterministic and add a button that can be clicked to trigger these changes? Not necessary, but if you go with this button approach you could probably combine `view_to_component_request`, `view_to_component_args`, and `view_to_component_kwargs` together into one component with separate buttons to trigger each.

Reply:
We aren't depending on the 3 second delays. I just added them as a "human readable" way to tell if the test is working.

Should I completely remove the delay?

Also I think separate views is more appropriate. Pytest styling conventions say all tests should be written out long-form and not combined (Always `WET`, never `DRY`). It helps improve readability and traceability of exception stacks.